carbon-design-system/carbon-addons-iot-react
# fix(dashboardgrid): should not set state if component unmounts

Closes #1577

**Summary**

- The setShouldAnimate call was happening after the component unmounted in some cases.  This cancels the function if the component unmounts

**Change List (commits, features, bugs, etc)**

- fix(DashboardGrid): cancel function and regen GridLayout when the bit changes

fixes this problem
```
onsole.error
      Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
          in DashboardGrid (created by DashboardEditor)
          in ErrorBoundary (created by DashboardEditor)
          in div (created by DashboardEditor)
          in div (created by DashboardEditor)
          in div (created by DashboardEditor)
          in div (created by DashboardEditor)
```

**Acceptance Test (how to verify the PR)**

- Verify that in the Dashboard stories, if isEditable is set, then it's still draggable.
- Verify in non editable, it's not draggable
